apps/emqx_durable_storage/src/emqx_ds_storage_layer.erl

```diff
@@ -606,16 +600,20 @@ rocksdb_open(Shard, Options) ->
     ],
     case rocksdb:open(DBDir, DBOptions, ColumnFamilies) of
         {ok, DBHandle, [_CFDefault | CFRefs]} ->
-            persistent_term:put({?MODULE, Shard, data_dir}, DataDir),
             {CFNames, _} = lists:unzip(ExistingCFs),
             {ok, DBHandle, lists:zip(CFNames, CFRefs)};
         Error ->
             Error
     end.
 
--spec db_dir(file:filename(), shard_id()) -> file:filename().
-db_dir(BaseDir, {DB, ShardId}) ->
-    filename:join([BaseDir, atom_to_list(DB), binary_to_list(ShardId)]).
+-spec db_dir(shard_id()) -> file:filename().
+db_dir({DB, ShardId}) ->
+    filename:join([base_dir(DB), atom_to_list(DB), binary_to_list(ShardId)]).
+
+-spec base_dir(emqx_ds:db()) -> file:filename().
+base_dir(DB) ->
```

Comment:
For starters, can we have a simpler config that doesn't take database into consideration, to simplify the schema? I feel like trying to move different DBs to different places is just going to confuse the users and add more variables for debugging.

We can add the ability to override the configs for the particular databases later, if real world requirements dictate so.